force-app/main/default/lwc/donationHistoryTable/donationHistoryTable.js

```diff
@@ -63,16 +71,13 @@ export default class DonationHistoryTable extends LightningElement {
         this._filter = value;
         this.retrieveDonationHistory();
         this.infiniteScroll = true;
-    }
-    
-    connectedCallback() {
-        this.retrieveDonationHistory();
+        this.tableElement ? this.tableElement.enableInfiniteLoading = true : null;
     }
 
     // eslint-disable-next-line @lwc/lwc/no-async-await
-    async retrieveDonationHistory() {
-        getDonationHistory({contactId: this.contactId, offset: 0, filter : this.filter})
-        .then(data => {
+    async retrieveDonationHistory(){
```

Comment:
This function does not need to be `async` if we're using promises inside of the function body instead of `await`.